src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/x64/sysv/CallArranger.java

```diff
@@ -119,7 +120,7 @@ public static Bindings getBindings(MethodType mt, FunctionDescriptor cDesc, bool
         if (!forUpcall) {
             //add extra binding for number of used vector registers (used for variadic calls)
             csb.addArgumentBindings(long.class, MemoryLayouts.SysV.C_LONG,
-                    List.of(new Binding.Move(rax, long.class)));
+                    Binding.builder().move(rax, long.class).build());
```

Comment:
can't you use the factory here?

Reply:
Good catch. I started out doing everything with the builder, but if we have both, this should use the factory.